File: 2021/01/31/beomjo/algorithm/BOJ/java/1158.java
import java.io.*;
import java.util.LinkedList;
import java.util.Queue;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int k = Integer.parseInt(input[1]);

        Queue<Integer> queue = new LinkedList<>();
        for (int i = 1; i <= n; i++) {
            queue.offer(i);
        }

        StringBuilder sb = new StringBuilder();

        sb.append("<");
        while (queue.size() > 1) {
            for (int j = 1; j < k; j++) {
                queue.offer(queue.poll());
            }

            sb.append(queue.poll()).append(", ");
        }
        sb.append(queue.poll()).append(">");
        System.out.print(sb);
    }
}

File: 2021/01/31/beomjo/algorithm/BOJ/kotlin/1158.kt
import java.util.*

fun main () {
    val scanner = Scanner(System.`in`)
    val n = scanner.nextInt()
    val k = scanner.nextInt()

    val queue = LinkedList((1..n).toMutableList())

    print("<")
    while (queue.size > 1) {
        repeat((1 until k).count()) { queue.add(queue.pop()) }
        print("${queue.pop()}, ")
    }
    print("${queue.remove()}>")
}